### PR TITLE
DiskRP swagger changes for TVM and CVM of version 2021-08-01

### PR DESCRIPTION
**Pull request by anshulsolanki21**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by haagha** [comment]
@ArcturusZhang can we suppress spellcheck for secureVMdiskEncryptionSetId?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json [hunk 1]
```diff
@@ -2814,6 +2842,14 @@
...
                 "value": "CopyStart",
                 "description": "Create a new disk by using a deep copy process, where the resource creation is considered complete only after all data has been copied from the source."
+              },
+              {
+                "value": "ImportSecure",
+                "description": "Similar to Import create option. Create a new TrustedLaunch or ConfidentialVM supported disk by importing additional VM guest state blob specified by securityDataUri in storage account specified by storageAccountId"
+              },
+              {
+                "value": "UploadPreparedSecure",
+                "description": "Similar to Upload create option. Create a new TrustedLaunch or ConfidentialVM supported disk and upload using write token in both disk and VM guest state blob"
```
Comment (lakmeedee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json:2852]: "Similar to Upload create option. Create a new TrustedLaunch or ConfidentialVM supported disk and upload using write token in both disk and VM guest state blob"
should be modified as
"Similar to Upload create option. Create a new Trusted Launch VM or Confidential VM supported disk and upload using write SAS token to both disk and VM guest state blob"

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json [hunk 2]
```diff
@@ -2853,6 +2889,10 @@
           "type": "integer",
           "format": "int32",
           "description": "Logical sector size in bytes for Ultra disks. Supported values are 512 ad 4096. 4096 is the default."
+        },
+        "securityDataUri": {
+          "type": "string",
+          "description": "If createOption is ImportSecure, this is the URI of a VM guest state blob to be imported."
```
Comment (lakmeedee) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json:2895]: What about UploadPreparedSecure Create Option?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json (hunk 1 above)
Comment (haagha) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json:2852]: @anshulsolanki21

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json [hunk 3]
```diff
@@ -2935,6 +2978,18 @@
...
             "value": "TrustedLaunch",
             "description": "Trusted Launch provides security features such as secure boot and virtual Trusted Platform Module (vTPM)"
+          },
+          {
+            "value": "ConfidentialVM_VMGuestStateOnlyEncryptedWithPlatformKey",
+            "description": "Indicates Confidential VM disk with only VM guest state blob encrypted"
+          },
+          {
+            "value": "ConfidentialVM_DiskEncryptedWithPlatformKey",
+            "description": "Indicates Confidential VM disk with both OS disk blob and VM guest state blob encrypted with a platform managed key"
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json:2988]: don't mention blobs.. both Os disk and VM guest state encrypted

Comment (chetmaddula) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json:2988]: same for all other additions.. remove any references to blob

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json [hunk 4]
```diff
@@ -2979,6 +3042,11 @@
           "readOnly": true,
           "type": "string",
           "description": "A SAS uri for accessing a disk."
+        },
+        "securityDataAccessSAS": {
+          "readOnly": true,
+          "type": "string",
+          "description": "A SAS uri for accessing a VM guest state blob."
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json:3049]: remove blob

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json [hunk 5]
```diff
@@ -2498,6 +2520,10 @@
           {
             "value": "EncryptionAtRestWithPlatformAndCustomerKeys",
             "description": "Resource using diskEncryptionSet would be encrypted at rest with two layers of encryption. One of the keys is Customer managed and the other key is Platform managed."
+          },
+          {
+            "value": "ConfidentialVmEncryptedWithCustomerKey",
+            "description": "Confidential VM supported disk encrypted with customer managed key."
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json:2526]: please be more descriptive similar to above. Would this encrypt only the guest state or can be used for disks too?

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json [hunk 6]
```diff
@@ -2476,6 +2493,10 @@
           {
             "value": "EncryptionAtRestWithPlatformAndCustomerKeys",
             "description": "Disk is encrypted at rest with 2 layers of encryption. One of the keys is Customer managed and the other key is Platform managed."
+          },
+          {
+            "value": "ConfidentialVmEncryptedWithCustomerKey",
+            "description": "Confidential VM supported disk encrypted with customer managed key."
```
Comment (chetmaddula) [inline specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json:2499]: same as below.. needs to be more descriptive. Please ask feature PM for pointers

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json (hunk 3 above)
Comment (anshulsolanki21) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json]: removed blob references:

also modified following descriptions -


{
                 "value": "ImportSecure",
-                "description": "Similar to Import create option. Create a new Trusted Launch VM or Confidential VM supported disk by importing additional VM guest state blob specified by securityDataUri in storage account specified by storageAccountId"
+                "description": "Similar to Import create option. Create a new Trusted Launch VM or Confidential VM supported disk by importing additional blob for VM guest state specified by securityDataUri in storage account specified by storageAccountId"
               },


"securityDataUri": {
           "type": "string",
-          "description": "If createOption is ImportSecure, this is the URI of a VM guest state blob to be imported."
+          "description": "If createOption is ImportSecure, this is the URI of a blob to be imported into VM guest state."
         }

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json (hunk 5 above)
Comment (anshulsolanki21) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json]: checked with PM, and modified to -
"Confidential VM supported disk and VM guest state would be encrypted with customer managed key."

File: specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json (hunk 6 above)
Comment (anshulsolanki21) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2021-08-01/disk.json]: checked with PM, and modified to -
"Confidential VM supported disk and VM guest state would be encrypted with customer managed key."

Also removed "ConfidentialVmEncryptedWithCustomerKey" from "EncryptionType" enum, as it is used at disk level. (not for DES type)

**Review by chetmaddula — APPROVED** [review]

**Pull request merged**